src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -439,10 +441,48 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 case 1:
                     assert(!instrIsRMW);
-                    if (targetReg != falseReg)
+
+                    if ((targetReg != falseReg))
                     {
-                        GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, falseReg);
+                        // If targetReg is not the same as `falseReg` then need to move
+                        // the `falseReg` to `targetReg`.
+
+                        if (op1->IsTrueAllMask())
+                        {
+                            // We already skip importing ConditionalSelect if op1 == trueAll, however
+                            // if we still see it here, it is because we wrapped the predicated instruction
```

Comment:
This isn't necessarily a safe presumption.

Importer will miss it in the case that `op1` is an inlined method call or a CSE that is later a propagated constant, etc.

Reply:
Ah i see, so I think, we should update the check to also include `intrin.op3->IsVectorZero()`? So basically if `(op1 == AllTrueMask) && (op3 == Zero))`, we just generate the instruction directly.